osbenchmark/worker_coordinator/runner.py

```diff
@@ -967,6 +968,75 @@ async def _scroll_query(opensearch, params):
                 "took": took
             }
 
+        async def _vector_search_query_with_recall(opensearch, params):
+            """
+            Perform vector search and report recall@k , recall@r and time taken to perform recall in ms as
+            meta object.
+            """
+            result = {
+                "weight": 1,
+                "unit": "ops",
+                "success": True,
+            }
+
+            def recall_at_r(results, neighbors, r):
+                """
+                Calculates the recall@R for a set of queries against a ground truth nearest
+                neighbor set
+                Args:
+                    results: list containing ids of results returned by OpenSearch.
+                    neighbors: set containing ids of the actual neighbors
+                    neighbors for a set of queries
+                    r: number of top results to check from the neighbors
+                Returns:
+                    Recall at R
+                """
+                correct = 0.0
+                if neighbors is None:
+                    return 0.0
```

Comment:
lets add a log message stating that neighbors are not found.

Reply:
Ack